### DISCUSSION TITLE
Dynamic window title and dynamic tab title that are different from each other

### DISCUSSION THREAD

**Discussion by porridgewithraisins** [kovidgoyal/kitty#7661] [Q&A] [answered]
I want the OS window title to be `Kitty: $(pwd)`, and the tab title to be `$(pwd): $(executing/last executed command)`

How do I achieve this?

I tinkered a bit but couldn't figure it out.

What I tried:
 - setting window title via escape codes in PS1
 - then setting tab title via `set-tab-title` in PROMPT_COMMAND
 - all this after setting `shell_integration no-title`

Platform: Linux
Kitty version: 0.35.2
Shell: bash

**Comment by kovidgoyal** [comment]
You cannot. Why would you want to waste space showing Kitty in your
titlebar anyway? If you want to match kitty os windows use the --class
and --name options.

**Reply by porridgewithraisins** [reply]
So basically I have a clipboard manager that does `xdotool getactivewindow getwindowname` and saves that to an index and lets me search copied items by window. I would like to search `kitty` and have it work. By default kitty puts the last run command/current directory as the window name (not sure when which of the two happens). I wanted to prepend "Kitty" to that. While I managed to achieve this, I realised that this also changes the tab title to have "Kitty" at the beginning, which I felt was redundant. So I wanted the behaviour described above (even if the currently running command is missing, that's ok, window name = Kitty, tab name = current working directory suffices)

**Reply by kovidgoyal** [reply]
Yeah but why do you need kitty in the window title for that? Do xdotool getactivewindow getclassname and prepend the class to the title.

**Reply by porridgewithraisins** [reply]
Hmm, that solves the problem at hand I guess.

That aside, is there a specific reason we can't decouple the tab and window titles though? I felt like it was pretty un-intuitive that I couldn't do that. Something like window_title_template alongside the existing tab_title_template, both decoupled, will be perfect.

**Comment by kovidgoyal** [comment, marked as answer]
On Wed, Jul 24, 2024 at 03:19:11AM -0700, Santhanam wrote:
> Hmm, that solves the problem at hand I guess.
>
> That aside, is there a specific reason we can't decouple the tab and window titles though? I felt like it was pretty un-intuitive that I couldn't do that. Something like window_title_template alongside the existing tab_title_template, both decoupled, will be perfect.

Feel free to send a PR for it, not something I find worth the effort.

**Reply by porridgewithraisins** [reply]
Cool, I'll see if I can work on it when I get the time. Thank you!

**Comment by chris4prez** [comment]
I see the merits in having both window and tab title decoupled and came here looking for a solution as I have a need for such functionality. 

Primarily in my use case focus/attention related, as I have bad focus/attention issues and distract too easily. Any extraneous changing items or text catches my eyes and moves me away from my prompt where I want to be or should be makes me lose focus...

I'd like to be able to keep my window title bar clear/blank and non-changing, however have my tabs denote where I left off. (i.e. path or app)

- Window Title I would want blank so there is less change happening and drawing my eyes off the line of commands where I want to focus. Every time it changes it catches me and I end up glancing up.

- Window Title I could care less for having path, as I tend to like it in my prompt, again for focus reasons to keep my eyes tracking the correct location where I'm working

- Tab title - I want dynamic to know where I last left off, not necessarily need the info in the active tab, but those tabs I left open it would be invaluable.

This way I stay organized, focused, and do not lose attention. This becomes even more valuable with OLED and HDR displays when the text is bright white when in dark mode and makes it super easy to catch the change occurring.

A simple config line to tweak both separately would be invaluable to people with similar challenges like me.
